### ISSUE TITLE
Overlap between IfcAdvancedBrep, IfcFacetedBrep, IfcTessellatedFaceSet and its subtypes

### ISSUE THREAD

**Issue by peterrdf** [buildingSMART/IFC_UML#46]
The following entities have some similarities:
   IfcAdvancedBrep
      IfcAdvancedBrepWithVoids
   IfcFacetedBrep
      IfcFacetedBrepWithVoids
   IfcTessellatedFaceSet
      IfcPolygonalFaceSet
      IfcTriangulatedFaceSet
Therefore the question is if some of these entities can be merged. IfcAdvancedBrep (IfcAdvancedBrepWithVoids) are clearly semantically the most rich concepts. If these entities would be replaced by the others semantic information will be lost, the most close here would be IfcPolygonalFaceSet. Also the other way around removing IfcPolygonalFaceSet or IfcTriangulatedFaceSet would prevent storing some of the basic geometry representations and a clean way to define individual triangle / face color assignment. 

IfcFacetedBrep allows advanced representations of how faces are build up, via curved polygons, trimmed polygons etc. as does its subtype entity supporting openings. Therefore this entity has extra value over tessellated representations. The other way around tessellated entities are able to represent knowledge in much more condense and easy manner to process.

The bottom line is that all these 7 entities have their own clear benefits and my opinion would be to keep them separate and not merge them.
